src/java/org/apache/cassandra/db/commitlog/EncryptedSegment.java

```diff
@@ -150,4 +149,28 @@ public long onDiskSize()
     {
         return lastWrittenPos;
     }
+
+    protected static class EncryptedSegmentBuilder extends CommitLogSegment.Builder<EncryptedSegment>
+    {
+
+        public EncryptedSegmentBuilder(AbstractCommitLogSegmentManager segmentManager)
+        {
+            super(segmentManager);
+        }
+
+        @Override
+        public EncryptedSegment build()
+        {
+            return new EncryptedSegment(segmentManager,
+                                        path ->  FileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE));
+        }
+
+        @Override
+        public SimpleCachedBufferPool createBufferPool()
+        {
+            return new SimpleCachedBufferPool(DatabaseDescriptor.getCommitLogMaxCompressionBuffersInPool(),
+                                              DatabaseDescriptor.getCommitLogSegmentSize(),
+                                              BufferType.ON_HEAP);
```

Comment:
The comment about the type of buffer
```
        // Note: we want to keep the compression buffers on-heap as we need those bytes for encryption,
        // and we want to avoid copying from off-heap (compression buffer) to on-heap encryption APIs
```
needs to be preserved here.